### ISSUE TITLE
"deno test" should run all the functions named test* in current directory

### ISSUE THREAD

**Issue by ry** [denoland/deno#20]
And only in files that end in `_test.ts`.

Example:
```
// foo_test.ts
function testBar() {
  if (1 != 2) {
    throw Error("Test failed.");
  }
}
```

**Comment by ry** [comment]
I think it’s better to have these thing prescribed - it’s very arbitrary and better if everyone just uses the same system

**Comment by yoavmmn** [comment]
I think it's good if everyone uses the same system, but we shouldn't force the use of the `test*` pattern. Maybe it should be the default, but it will be better allowing custom pattern like `deno test tests/*`.

**Comment by robbym** [comment]
Attempting to implement this: https://github.com/robbym/deno/commit/d4b66fb9a8e0ce1ddd8b5c14f60a2b34c9bda1f8

How does one get stuff, like all the function names in a file, from the C API?
Functionality not there yet?

**Comment by ry** [comment]
Yeah functionality is not there yet. Hold off on this for a bit... We've got one more sprint and then we'll be up and running on the prototype's runtime. It'll make more sense to talk about it once we've got that working.

**Comment by hayd** [comment]
So, you can do this if the function is exported:
```ts
// foo_test.ts
export function testFoo() {
  console.log("foo");
}

// test.ts
(async () => {
  const name = "foo_test.ts";
  const t = await import(name);
  t.testFoo();
})();
```

Is there a way to access these if the function is not exported?

**Comment by bartlomieju** [comment]
@ry close in favor of testing module in standard library?

**Comment by hayd** [comment]
Once `import()` is working I will add this to std testing similar to --fmt.

**Comment by ry** [comment]
This is still desired. @hayd looking forward to it.

**Comment by bartlomieju** [comment]
Resolved in #2783!

CC @ry

**Comment by ry** [comment]
Nice work :D
